src/librustc_typeck/check/wfcheck.rs

```diff
@@ -111,15 +111,14 @@ pub fn check_item_well_formed(tcx: TyCtxt<'_>, def_id: DefId) {
                 }
                 ty::ImplPolarity::Negative => {
                     // FIXME(#27579): what amount of WF checking do we need for neg impls?
```

Comment:
I think that if we opt for the plan that I have vaguely in mind, then we would wind up doing the same WF checking on negative impls that we do on positive impls. Effectively, `Trait` and `!Trait` become two "traits" which are not permitted to overlap.

This would be a good thing to add to the tracking issue as "to be resolved".

Reply:
That makes sense. Thanks for adding it to the issue.